tests/Happyr/LinkedIn/LinkedInTest.php

```diff
@@ -35,6 +36,19 @@ public function testConstructor()
             'Expect the API secret to be set.');
     }
 
+    public function testInitProperlyAssignsPassedInStorage()
```

Comment:
Hm, you are correct but you have one flaw in your statement. 
> $this->assertEquals($sessionStorage, $ln->getStorage()); is more prudent as it will make sure that the actual variable we passed to the constructor got assigned to the member variable.

This is not true. See the following example: 
```php 
// IlluminateSessionStorage extends SessionStorage
        $a = new IlluminateSessionStorage();
        $b = new SessionStorage();

        $this->assertEquals($a, $b); //false

        $c = clone $b;
        $this->assertEquals($b, $c); //true

        $c->foo = 'bar';
        $this->assertEquals($b, $c); //false

        $d = new IlluminateSessionStorage();
        $this->assertEquals($a, $d); //true
```
Take a look at the very last line. It shows that `assertEquals` compare if the objects looks the same, but not if the variables reference the same object. 

Do as you suggested. (Use the original assert but with IlluminateSessionStorage.) It will be great.